modules/jenkins-plugin/src/main/java/com/smartbear/ready/jenkins/RemoteTestStarter.java

```diff
@@ -0,0 +1,539 @@
+        this.testSuiteName = testSuiteName;
+        this.repositoryName = repositoryName;
+        this.environment = environment;
+        this.hostAndPort = hostAndPort;
+        this.tags = tags;
+        this.projectFilePassword = projectFilePassword;
+        this.projectProperties = projectProperties;
+        this.extraFiles = extraFiles;
+        this.reportFile = reportFile;
+    }
+
+    private TestType testTypeForString(String typeString) {
+        return Arrays.stream(TestType.values()).filter(type -> type.name().equals(typeString))
+                .findFirst()
+                .orElse(TestType.LOCAL);
+    }
+
+    @Override
+    public boolean perform(AbstractBuild build, Launcher launcher, BuildListener listener) throws AbortException {
+        try {
+            workspace = new File(build.getWorkspace().toURI());
+        } catch (Exception e) {
+            throw new AbortException("Could not read path to workspace");
+        }
+        try {
+            List<TestServerExecution> execution = testType == TestType.REPOSITORY ? runRepositoryTests() :
+                    runTestFiles(build);
+            log.info("Finished TestServer execution - {} projects executed", execution.size());
+        } catch (IOException e) {
+            log.error("TestServer API call failed", e);
+            throw new AbortException("Could not start tests from project " + pathToProjectFile + " on server " + serverUrl);
+        } catch (ApiException e) {
+            log.error("TestServer API call failed when adding files", e);
+            throw new AbortException("Failed to attached dependent files to the project " + pathToProjectFile);
+        } catch (Exception e) {
+            String message = "Unexpected error when running " + pathToProjectFile;
+            log.error(message, e);
+            throw new AbortException(message + ": " + e.getMessage());
+        }
+        return true;
+    }
+
+    private List<TestServerExecution> runRepositoryTests() throws IOException, ApiException {
+        TestServerClient client = makeTestServerClient();
+        TestServerExecution repositoryExecution = executeRepositoryProject(client);
+        String repositoryProjectPath = repositoryName + "/" + pathToProjectFile;
+        repositoryExecution = addFilesIfExecutionIsPending(client, repositoryProjectPath, repositoryExecution);
+        return Collections.singletonList(repositoryExecution);
+    }
+
+    private Collection<File> getFilesToRun(File targetFile) throws AbortException {
+        if (targetFile.isDirectory()) {
+            String extension = testType == TestType.LOCAL_RECIPE ? ".json" : ".xml";
+            File[] filesInDirectory = targetFile.listFiles((dir, name) -> name.endsWith(extension));
+            if (filesInDirectory == null) {
+                throw new AbortException("No project files or recipes found in path " + pathToProjectFile);
+            } else {
+                return Arrays.asList(filesInDirectory);
```

Comment:
No, you won't get here then. This code will only run if the path ends in a wildcard.